lib/googleauth/bearer_token.rb

```diff
@@ -0,0 +1,157 @@
+    #
+    class BearerTokenCredentials
+      include Google::Auth::BaseClient
+
+      # @private Authorization header name
+      AUTH_METADATA_KEY = Google::Auth::BaseClient::AUTH_METADATA_KEY
+
+      # @private Allowed token types
+      ALLOWED_TOKEN_TYPES = [:access_token, :jwt, :id_token, :bearer_token].freeze
+
+      # @return [String] The token to be sent as a part of Bearer claim
+      attr_reader :token
+      # The following aliasing is needed for BaseClient since it sends :token_type
+      alias access_token token
+      alias jwt token
+      alias id_token token
+      alias bearer_token token
+
+      # @return [Time, nil] The token expiry time provided by the end-user.
+      attr_reader :expiry
+
+      # @return [Symbol] The token type. Allowed values are
+      #   :access_token, :jwt, :id_token, and :bearer_token.
+      attr_reader :token_type
+
+      # @return [String] The universe domain of the universe
+      #   this token is for
+      attr_accessor :universe_domain
+
+      class << self
+        # Create the BearerTokenCredentials.
+        #
+        # @param [Hash] options The credentials options
+        # @option options [String] :token The bearer token to use.
+        # @option options [Time, Numeric, nil] :expiry The token expiry time provided by the end-user.
+        #   Optional, for the end-user's convenience. Can be a Time object, a number of seconds since epoch.
+        #   If the expiry is `nil`, it is treated as "token never expires".
+        # @option options [Symbol] :token_type The token type. Allowed values are
+        #   :access_token, :jwt, :id_token, and :bearer_token. Defaults to :bearer_token.
+        # @option options [String] :universe_domain The universe domain of the universe
+        #   this token is for (defaults to googleapis.com)
+        # @return [Google::Auth::BearerTokenCredentials]
+        def make_creds options = {}
+          new options
+        end
+      end
+
+      # Initialize the BearerTokenCredentials.
+      #
+      # @param [Hash] options The credentials options
+      # @option options [String] :token The bearer token to use.
+      # @option options [Time, Numeric, nil] :expiry The token expiry time provided by the end-user.
+      #   Optional, for the end-user's convenience. Can be a Time object, a number of seconds since epoch.
+      #   If the expiry is `nil`, it is treated as "token never expires".
+      # @option options [Symbol] :token_type The token type. Allowed values are
+      #   :access_token, :jwt, :id_token, and :bearer_token. Defaults to :bearer_token.
+      # @option options [String] :universe_domain The universe domain of the universe
+      #   this token is for (defaults to googleapis.com)
+      def initialize options = {}
+        raise ArgumentError, "Bearer token must be provided" if options[:token].nil? || options[:token].empty?
+        @token = options[:token]
+        @expiry = if options[:expiry].is_a? Time
```

Comment:
nit: `case options[:expiry]` is more idiomatic.

Reply:
Ack